### ISSUE TITLE
Watermark detection across multiple pages

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/despeck#4]
In some cases, a watermark is placed on different locations and/or different pages -- for example, a watermark if placed on a blank page will give the full specification of the watermark, which can be extracted and used to remove watermarks on other pages (e.g., when a watermark obstructed by text).

How do we best support this:

1. Ask the user to specify a full watermark (i.e., watermark placed on a blank page) 
2. Automatically detect the best watermark amongst the series of pages, allow user to extract the watermark (as an image or specification) for future use?

Here's a sample attached.
<img width="741" alt="07" src="https://user-images.githubusercontent.com/11865/36974006-124c1414-20b0-11e8-8feb-fc37c81a0158.png">
<img width="744" alt="10" src="https://user-images.githubusercontent.com/11865/36974026-21096cea-20b0-11e8-8808-12287cf65e75.png">

**Comment by ronaldtse** [comment]
Blank watermark.
<img width="744" alt="04" src="https://user-images.githubusercontent.com/11865/36974985-9b37da44-20b3-11e8-838e-a9fc61289bbe.png">

**Comment by nattfodd** [comment]
Auto detection will fail in case of such tiny watermark, but the user is able to provide approx color of watermark (in this case simply `0000FF` - for primary blue channel), and it will be removed successfully.

**Comment by ronaldtse** [comment]
@nattfodd is it possible to detect the watermark since it is a common occurrence across pages?

I've found one more issue here is that the text resolution drops significantly after applying despeck.

e.g.
Before:

![screen shot 2018-03-07 at 2 04 49 am](https://user-images.githubusercontent.com/11865/37049502-fe1f2c18-21ab-11e8-97ea-cd180cf13bab.png)

After:
![screen shot 2018-03-07 at 2 04 55 am](https://user-images.githubusercontent.com/11865/37049501-fdd58d7e-21ab-11e8-980a-e224a955e4a1.png)

**Comment by nattfodd** [comment]
@ronaldtse even though we have watermark image on the blank page, it's not that easy to extract it  from the pages with text, because watermark is on a different place and/or with a different align. That makes the task very complicated comparing to color extraction, since we need to apply every possible transformations of watermark to the image...

**Comment by ronaldtse** [comment]
I wonder if there's an algorithm that can detect a common instance of sub-image "x" from a series of pages? Or, if we allow people to manually extract a sample watermark, provide it as input, and despeck can remove it?

**Comment by nattfodd** [comment]
If watermark position is still on every page, I believe we can extract it without touching the rest of image. But as I understand the story, the position may vary, isn't it?

**Comment by ronaldtse** [comment]
Yes, it will often move. It would be the detection of a common re-occuring image instance within a series of images.

https://stackoverflow.com/questions/29356639/detecting-all-occurrences-of-subimage-within-image
https://github.com/johnoneil/subimage
https://cvml.ist.ac.at/papers/lampert-iccv2009.pdf

**Comment by anieshvishnu** [comment]
Can this code differentiate between watermark and text in this pdf given below. The watermark and the text looks familiar.

[rest3.pdf](https://github.com/riboseinc/despeck/files/2629497/rest3.pdf)
